Userland/Libraries/LibRegex/RegexParser.cpp

```diff
@@ -460,6 +461,11 @@ bool PosixBasicParser::parse_nonduplicating_re(ByteCode& bytecode, size_t& match
 {
     // nondupl_RE : one_char_or_coll_elem_RE | Back_open_paren RE_expression Back_close_paren | BACKREF
     if (try_skip("\\(")) {
+        TemporaryChange change { m_current_capture_group_depth, m_current_capture_group_depth + 1 };
+        // Max number of addressable capture groups is 10, let's just be lenient
+        // and accept 20; anything past that is probably a silly pattern anyway.
+        if (m_current_capture_group_depth > 20)
+            return set_error(Error::InvalidPattern);
```

Comment:
I was gonna say we could `dbgln()` here, but I guess if someone ever hits this, they can go looking for whoever said the pattern was invalid. 🤷🏻

Reply:
Note that this is a BRE, there are only 10 addressable capture groups.
If someone actually _nests_ 20 groups in a BRE...they probably need other kinds of help